### PR TITLE
[run-gp] experiment with adding run-gp layer to subset of users

### PR DESCRIPTION
**Pull request by andreafalzetti**
## Description
This PR introduces a way to add a docker layer containing [run-gp](github.com/gitpod-io/run-gp) wrapped by a shell cmd `gp-run`. 

Co-Authored: @akosyakov @andreafalzetti @felladrin 

## Related Issue(s)

Fixes #

## How to test
1. Start a workspace and verify that `gp-run` is not available as a command
2. Add your email or userId in ConfigCat and run the test again
3. Start a workspace and check that `gp-run` is available and starts `run-gp`

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@andreafalzetti: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/ide-service/pkg/server/server.go [hunk 1]
```diff
@@ -338,6 +339,21 @@ func (s *IDEServiceServer) ResolveWorkspaceConfig(ctx context.Context, req *api.
 			resp.IdeImageLayers = append(resp.IdeImageLayers, desktopImageLayer)
 			resp.IdeImageLayers = append(resp.IdeImageLayers, userImageLayers...)
 		}
+
+		if ideConfig.GpRunImage != "" {
+
+			client := experiments.NewClient()
```
Comment (akosyakov) [inline components/ide-service/pkg/server/server.go:345]: only one client per a server, not for each request

**Review by akosyakov — APPROVED** [review]
lgtm, i did not try, rely on testing from @andreafalzetti 

/hold

we need squash commit and add co-author etc. unhold when you are ready

**Pull request merged**